### ISSUE TITLE
[3.5.1] Physics and code behaviour act differently at each scene load

### ISSUE THREAD

**Issue by belzecue** [godotengine/godot#71847]
### Godot version

3.5.1.stable

### System information

Linux Ubunt 20.4

### Issue description

I have a small 2d platform test project where I use Engine.get_physics_interpolation_fraction() to get smooth visualbody movement at any physics rate.  In the video demo below, which is me repeatedly running a Linux build of my project, you can see that whenever the player (a rigidbody) falls off screen, after five seconds the scene gets reloaded with get_tree().reload_current_scene().  However, I get different behaviours at each scene reload, which should not be happening.  I know Godot physics (bullet/godotphysics) is non-deterministic, but there's something else going on here.  There is no input during the recording of this video.  Everything is happening under Godot's internal physics alone.  There are no autoloads that are surviving a scene restart.  There are no resources holding global values.  I can't figure out why I'm getting different behaviour each time.

[Video demo here](https://www.youtube.com/watch?v=Gt4DZyaNRHI) (too big to attach to issue)

### Steps to reproduce

Load the MRP, make a build, and repeatedly run/kill the build to see if the starting behaviour is different.  Also observe that for many runs, each time the scene reloads, the syncing of the player visualbody (icon.png) and the player physicsbody (blue circle) lags a little more, as if there's an accumulating error with each reload.  You can see the accumulation of lag [during this run](https://youtu.be/Gt4DZyaNRHI?t=299) of the build, in the video

### Minimal reproduction project

Project code attached.

[physics_issue.zip](https://github.com/godotengine/godot/files/10474054/physics_issue.zip)

**Comment by lawnjelly** [comment]
It looks like you misunderstand how to do interpolation. If you study https://github.com/lawnjelly/smoothing-addon it should be helpful.

Briefly, if you store in `move2d.gd`:
```
var prev_tr : Transform2D
var curr_tr : Transform2D
```
Then in physics tick:
```
func _physics_process(delta: float) -> void:
	prev_tr = curr_tr
	curr_tr = get_parent().global_transform
```
Then in process:
```
	visual_body.global_transform = prev_tr.interpolate_with(
		curr_tr,
		Engine.get_physics_interpolation_fraction() 
	)
```

**Comment by lawnjelly** [comment]
Closing as not an engine bug.

**Comment by belzecue** [comment]
Thanks, @lawnjelly .  I certainly did misunderstand -- unfortunately, something I've prone to do with Godot.  (Last time was a total misunderstanding of call_deferred.)

I added the code to capture last and current rigidbody positions and now all working as expected.

The weird thing is, quite a while ago I did the same thing in a [3D implementation](https://github.com/belzecue/godot-physics-fraction/blob/master/Scripts/RigidBodyPhysicsTween.gd) and it worked seemingly perfectly, which is why I assumed I'd just do the same approach in 2D.  Obviously going to fix that 3D code right now.
